{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA_DIR = \"data\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/c/Users/kprosvir/Desktop/masters/env/lib/python3.6/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "meta_table = pd.read_table(os.path.join(DATA_DIR, \"metadata.tsv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_fold_change = meta_table[meta_table[\"Output type\"] == \"fold change over control\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['File accession', 'File format', 'Output type', 'Experiment accession',\n",
       "       'Assay', 'Biosample term id', 'Biosample term name', 'Biosample type',\n",
       "       'Biosample organism', 'Biosample treatments',\n",
       "       'Biosample treatments amount', 'Biosample treatments duration',\n",
       "       'Biosample genetic modifications methods',\n",
       "       'Biosample genetic modifications categories',\n",
       "       'Biosample genetic modifications targets',\n",
       "       'Biosample genetic modifications gene targets',\n",
       "       'Biosample genetic modifications site coordinates',\n",
       "       'Biosample genetic modifications zygosity', 'Experiment target',\n",
       "       'Library made from', 'Library depleted in', 'Library extraction method',\n",
       "       'Library lysis method', 'Library crosslinking method',\n",
       "       'Library strand specific', 'Experiment date released', 'Project',\n",
       "       'RBNS protein concentration', 'Library fragmentation method',\n",
       "       'Library size range', 'Biological replicate(s)', 'Technical replicate',\n",
       "       'Read length', 'Mapped read length', 'Run type', 'Paired end',\n",
       "       'Paired with', 'Derived from', 'Size', 'Lab', 'md5sum', 'dbxrefs',\n",
       "       'File download URL', 'Assembly', 'Genome annotation', 'Platform',\n",
       "       'Controlled by', 'File Status', 'Audit WARNING',\n",
       "       'Audit INTERNAL_ACTION', 'Audit NOT_COMPLIANT', 'Audit ERROR'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_table.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File accession</th>\n",
       "      <th>File format</th>\n",
       "      <th>Output type</th>\n",
       "      <th>Experiment accession</th>\n",
       "      <th>Assay</th>\n",
       "      <th>Biosample term id</th>\n",
       "      <th>Biosample term name</th>\n",
       "      <th>Biosample type</th>\n",
       "      <th>Biosample organism</th>\n",
       "      <th>Biosample treatments</th>\n",
       "      <th>...</th>\n",
       "      <th>File download URL</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Genome annotation</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Controlled by</th>\n",
       "      <th>File Status</th>\n",
       "      <th>Audit WARNING</th>\n",
       "      <th>Audit INTERNAL_ACTION</th>\n",
       "      <th>Audit NOT_COMPLIANT</th>\n",
       "      <th>Audit ERROR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENCFF409RWY</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>signal p-value</td>\n",
       "      <td>ENCSR571IIS</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0003042</td>\n",
       "      <td>H1-hESC</td>\n",
       "      <td>cell line</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF409RW...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>mild to moderate bottlenecking</td>\n",
       "      <td>NaN</td>\n",
       "      <td>insufficient read depth, poor library complexi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENCFF655FOD</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>fold change over control</td>\n",
       "      <td>ENCSR571IIS</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0003042</td>\n",
       "      <td>H1-hESC</td>\n",
       "      <td>cell line</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF655FO...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>mild to moderate bottlenecking</td>\n",
       "      <td>NaN</td>\n",
       "      <td>insufficient read depth, poor library complexi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENCFF669LJR</td>\n",
       "      <td>bed narrowPeak</td>\n",
       "      <td>peaks</td>\n",
       "      <td>ENCSR571IIS</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0003042</td>\n",
       "      <td>H1-hESC</td>\n",
       "      <td>cell line</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF669LJ...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>mild to moderate bottlenecking</td>\n",
       "      <td>NaN</td>\n",
       "      <td>insufficient read depth, poor library complexi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENCFF945OSD</td>\n",
       "      <td>bed narrowPeak</td>\n",
       "      <td>peaks</td>\n",
       "      <td>ENCSR571IIS</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0003042</td>\n",
       "      <td>H1-hESC</td>\n",
       "      <td>cell line</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF945OS...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mild to moderate bottlenecking</td>\n",
       "      <td>NaN</td>\n",
       "      <td>insufficient read depth, poor library complexi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENCFF297WYL</td>\n",
       "      <td>bigBed narrowPeak</td>\n",
       "      <td>peaks</td>\n",
       "      <td>ENCSR571IIS</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0003042</td>\n",
       "      <td>H1-hESC</td>\n",
       "      <td>cell line</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF297WY...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mild to moderate bottlenecking</td>\n",
       "      <td>NaN</td>\n",
       "      <td>insufficient read depth, poor library complexi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 52 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  File accession        File format               Output type  \\\n",
       "0    ENCFF409RWY             bigWig            signal p-value   \n",
       "1    ENCFF655FOD             bigWig  fold change over control   \n",
       "2    ENCFF669LJR     bed narrowPeak                     peaks   \n",
       "3    ENCFF945OSD     bed narrowPeak                     peaks   \n",
       "4    ENCFF297WYL  bigBed narrowPeak                     peaks   \n",
       "\n",
       "  Experiment accession     Assay Biosample term id Biosample term name  \\\n",
       "0          ENCSR571IIS  ChIP-seq       EFO:0003042             H1-hESC   \n",
       "1          ENCSR571IIS  ChIP-seq       EFO:0003042             H1-hESC   \n",
       "2          ENCSR571IIS  ChIP-seq       EFO:0003042             H1-hESC   \n",
       "3          ENCSR571IIS  ChIP-seq       EFO:0003042             H1-hESC   \n",
       "4          ENCSR571IIS  ChIP-seq       EFO:0003042             H1-hESC   \n",
       "\n",
       "  Biosample type Biosample organism  Biosample treatments  ...  \\\n",
       "0      cell line       Homo sapiens                   NaN  ...   \n",
       "1      cell line       Homo sapiens                   NaN  ...   \n",
       "2      cell line       Homo sapiens                   NaN  ...   \n",
       "3      cell line       Homo sapiens                   NaN  ...   \n",
       "4      cell line       Homo sapiens                   NaN  ...   \n",
       "\n",
       "                                   File download URL  Assembly  \\\n",
       "0  https://www.encodeproject.org/files/ENCFF409RW...      hg19   \n",
       "1  https://www.encodeproject.org/files/ENCFF655FO...      hg19   \n",
       "2  https://www.encodeproject.org/files/ENCFF669LJ...      hg19   \n",
       "3  https://www.encodeproject.org/files/ENCFF945OS...      hg19   \n",
       "4  https://www.encodeproject.org/files/ENCFF297WY...      hg19   \n",
       "\n",
       "   Genome annotation  Platform  Controlled by  File Status  \\\n",
       "0                NaN       NaN            NaN     archived   \n",
       "1                NaN       NaN            NaN     archived   \n",
       "2                NaN       NaN            NaN     archived   \n",
       "3                NaN       NaN            NaN     released   \n",
       "4                NaN       NaN            NaN     released   \n",
       "\n",
       "                    Audit WARNING  Audit INTERNAL_ACTION  \\\n",
       "0  mild to moderate bottlenecking                    NaN   \n",
       "1  mild to moderate bottlenecking                    NaN   \n",
       "2  mild to moderate bottlenecking                    NaN   \n",
       "3  mild to moderate bottlenecking                    NaN   \n",
       "4  mild to moderate bottlenecking                    NaN   \n",
       "\n",
       "                                 Audit NOT_COMPLIANT Audit ERROR  \n",
       "0  insufficient read depth, poor library complexi...         NaN  \n",
       "1  insufficient read depth, poor library complexi...         NaN  \n",
       "2  insufficient read depth, poor library complexi...         NaN  \n",
       "3  insufficient read depth, poor library complexi...         NaN  \n",
       "4  insufficient read depth, poor library complexi...         NaN  \n",
       "\n",
       "[5 rows x 52 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "env",
   "language": "python",
   "name": "env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
